src/cmd_index.cpp

```diff
@@ -86,9 +86,33 @@ static void set_field( library & lib, std::string const & name, std::string cons
     lib[ name ].front() = value;
 }
 
+static bool has_meta_file( std::string const & path )
+{
+    return fs_exists( path + "/meta/libraries.json" ) ||
+           fs_exists( path + "/.boost" );
+}
+
+static std::string find_meta_file( std::string const & path )
+{
+    bool meta_libraries = fs_exists(path + "/meta/libraries.json");
+    bool dot_boost = fs_exists(path + "/.boost");
+
+    if( meta_libraries && dot_boost )
```

Comment:
Oh, hey, I just realized that it's still going to work as done right now, it's just going to emit a warning message. In other words, if both files are present, we print a warning message and prefer the old version. Otherwise, we print a warning message and return `"meta/libraries.json"` (which is invalid, but if neither of these files are there we're already in trouble in the original code).